### PR TITLE
Adds litlogger integration

### PR DESCRIPTION
**Pull request by justusschock**
## What does this PR do?

Integrates LitLogger as a first-class logger to pytorch-lightning.

Users will be able to log directly to lightning.ai from pytorch_lightning without any additional setup.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21430.org.readthedocs.build/en/21430/

### CODE REVIEW

File: src/lightning/pytorch/loggers/litlogger.py [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+            store_step: Whether to store the step field with each logged value.
+            log_model: If True, automatically log model checkpoints as artifacts.
+            save_logs: If True, capture and upload terminal logs.
+            checkpoint_name: Override the base name for logged checkpoints.
+
+        Example::
+
+            from lightning.pytorch import Trainer
+            from lightning.pytorch.demos.boring_classes import BoringModel, BoringDataModule
+            from litlogger import LightningLogger
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
            from lightning.pytorch.loggers.litlogger import LitLogger
```

File: src/lightning/pytorch/loggers/litlogger.py [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+
+class LitLogger(Logger):
+    """Logger that streams metrics and artifacts to the Lightning.ai platform."""
+
+    LOGGER_JOIN_CHAR = "-"
+
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        root_dir: Optional[_PATH] = None,
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
        root_dir: Optional[_PATH] = None,
        name: Optional[str] = None,
```
It seems like we might need to change the order here a bit.

File: src/lightning/pytorch/loggers/litlogger.py [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+                def training_step(self, batch, batch_idx: int):
+                    loss = self.step(batch)
+                    # logging the computed loss
+                    self.log("train_loss", loss)
+                    return {"loss": loss}
+
+            trainer = Trainer(
+                max_epochs=10,
+                enable_model_summary=False,
+                logger=LightningLogger("./lightning_logs", name="boring_model")
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
                logger=LitLogger("./lightning_logs", name="boring_model")
```

File: src/lightning/pytorch/loggers/litlogger.py [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+        """Log a file as an artifact to the Lightning platform.
+
+        The file will be logged in the Teamspace drive,
+        under a folder identified by the experiment name.
+
+        Args:
+            path: Path to the file to log.
+
+        Example::
+            logger = LightningLogger(...)
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
            logger = LitLogger(...)
```

File: src/lightning/pytorch/loggers/litlogger.py [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+
+    def _scan_and_log_checkpoints(self, checkpoint_callback: ModelCheckpoint) -> None:
+        """Find new checkpoints from the callback and log them as model artifacts."""
+        checkpoints = _scan_checkpoints(checkpoint_callback, self._logged_model_time)
+
+        for timestamp, path_ckpt, _score, tag in checkpoints:
+            if not self._checkpoint_name:
+                self._checkpoint_name = self.experiment.name
+            # Ensure the version tag is unique by appending a timestamp
+            unique_tag = f"{tag}-{int(datetime.utcnow().timestamp())}"
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
            unique_tag = f"{tag}-{int(datetime.now(timezone.utc).timestamp())}"
```


The method `datetime.utcnow()` seems to have been deprecated in Python 3.12. According to the documentation, using `datetime.now(timezone.utc)` is the recommended alternative: https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

File: src/lightning/pytorch/loggers/litlogger.py [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+    @property
+    @override
+    def version(self) -> Optional[str]:
+        """Get the experiment version - its time of creation."""
+        return self._version
+
+    @property
+    @override
+    def root_dir(self) -> str:
+        """Gets the save directory where the TensorBoard experiments are saved."""
```
Comment (bhimrazy) [inline src/lightning/pytorch/loggers/litlogger.py]:
```suggestion
        """Gets the save directory where the litlogger experiments are saved."""
```

@justusschock , a quick question on the wording here — should we refer to these as litlogger experiments or Lightning.ai experiments?

**Review by williamFalcon — APPROVED** [review]
Been waiting 7 years for this lol

https://github.com/williamFalcon/test-tube
